src/csharp/Microsoft.Spark/Broadcast.cs

```diff
@@ -0,0 +1,190 @@
+                    "org.apache.spark.api.python.PythonRDD",
+                    "readBroadcastFromFile",
+                    javaSparkContext,
+                    _path);
+                }
+                else
+                {
+                    SparkConf sparkConf = _sc.GetConf();
+                    sc._encryption_enabled = bool.Parse(
+                        sparkConf.Get("spark.io.encryption.enabled",
+                        "false"));
+
+                    _python_broadcast = (JvmObjectReference)sparkContext.Jvm.CallStaticJavaMethod(
+                        "org.apache.spark.api.python.PythonRDD",
+                        "setupBroadcast",
+                        _path);
+
+                    if (sc._encryption_enabled)
+                    {
+                        throw new NotImplementedException(
+                                "Broadcast encryption is not supported yet.");
+                    }
+                    else
+                    {
+                        WriteBroadcastValueToFile(value);
+                    }
+
+                    _jvmObject = (JvmObjectReference)sparkContext.Jvm.CallStaticJavaMethod(
+                        "org.apache.spark.sql.api.dotnet.SQLUtils",
+                        "createBroadcast",
+                        sparkContext,
+                        _python_broadcast);
+                }
+
+                _bid = (long)_jvmObject.Invoke("id");
+                BroadcastRegistry.s_listBroadcastVariables.Add(_jvmObject);
+            }
+            else
+            {
+                // We're on an executor.
+                _sc = null;
+                _jvmObject = null;
+                _python_broadcast = null;
+                
+                if (path != null)
+                {
+                    _path = path;
+                }
+                else
+                {
+                    throw new SystemException(
+                        "broadcast called from executor with path not set.");
+                }
+            }
+        }
+        JvmObjectReference IJvmObjectReferenceProvider.Reference => _jvmObject;
+
+        /// <summary>
+        /// Function that creates a file to store the broadcast value in the given path
+        /// </summary>
+        /// <param name="value">Broadcast value to be written to the file</param>
+        public void WriteBroadcastValueToFile(object value)
+        {
+            Directory.CreateDirectory(_sc._temp_dir);
+            FileStream f = File.Create(_path);
+            Dump(value, f);
+            f.Close();
+        }
+
+        /// <summary>
+        /// Function that serializes and stores the object passed to the given Stream
+        /// </summary>
+        /// <param name="value">Serializable object</param>
+        /// <param name="stream">Stream to which the object is serialized</param>
+        public void Dump(object value, Stream stream)
```

Comment:
why is this public?

Reply:
I'll make it private, isn't being called from anywhere else. Thanks!